lib/src/js/polymer_base_dart.html

```diff
@@ -91,10 +105,32 @@
         // If you call notifyPath from dart we need to do manual modifications
         // on any JsArray or JsObject instances.
         var parts = this._getPathParts(path);
-        if (parts.length > 1) {
-          var model = this.get(parts.splice(0, parts.length - 1));
-          if (model[parts[0]] != value) model[parts[0]] = value;
+        // if (parts.length > 1) {
+
+        var thisArg =  this;
+        for (var i=0;i<parts.length-1;i++) {
```

Comment:
In propertyChange too you get the modify path. How could be used by users of that API if it is not the original one?
Also note the splice info you get : the keys are not corresponding to the index that changed for the same reason. If one was going to use it how he can interpret those numbers? There is no way that they can be converted to indexes.